feat(e2e): generate branch with the updated e2e screenshots (#2915)
AmadeusITGroup/otter: docs/testing/VISUAL_TESTING_REPORTER.md

# Visual testing reporter

> When e2e visual testing fail, screenshots have to be updated.

Reviewer: ```suggestion
When the e2e visual testing fails, screenshots have to be updated.
```

> As it was a "long" process to setup a docker in local and to re-run the e2e tests in the same context as the CI.

Reviewer: ```suggestion
As it was a "long" process to setup a docker in local and to re-run the e2e tests in the same context as the CI,
```
or 
```suggestion
It was a "long" process to setup a docker in local and to re-run the e2e tests in the same context as the CI.
```

> As it was a "long" process to setup a docker in local and to re-run the e2e tests in the same context as the CI.

Reviewer 2: it's not that bad :(
I would say something like `To facilitate the update of the screenshots directly from the CI` or something like that

> We created a github action [AmadeusITGroup/otter/tools/github-actions/create-branch-e2e-screenshots](tools/github-actions/create-branch-e2e-screenshots/readme.md) so a branch will be created each time visual tests failed.

Reviewer: Depending on what you have decided to keep as a suggestion:
```suggestion
we created a github action [AmadeusITGroup/otter/tools/github-actions/create-branch-e2e-screenshots](tools/github-actions/create-branch-e2e-screenshots/readme.md) so a branch will be created each time visual tests failed.
```
or 
```suggestion
Hence, we created a github action [AmadeusITGroup/otter/tools/github-actions/create-branch-e2e-screenshots](tools/github-actions/create-branch-e2e-screenshots/readme.md) so a branch will be created each time visual tests failed.
```

For that, we had to create a [Playwright reporter](https://playwright.dev/docs/test-reporters) to retrieve the actual and expected screenshots paths when using [`toHaveScreenshot`](https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-screenshot-1)

## Usage

> You need to have the dependency to `@o3r/testing`.

Reviewer: ```suggestion
** Prerequisite **
- Install `@o3r/testing` as a dev-dependency.
```

> Then in your Playwright configuration, you can specify `reporter`.

Reviewer: ```suggestion
Update the `reporter` in your Playwright configuration.
```

```typescript
export default defineConfig({
  reporter: '@o3r/testing/visual-testing-reporter',
});
```

In case you use Yarn PnP, you have to resolve the path to the reporter.
```typescript
export default defineConfig({
  reporter: require.resolve('@o3r/testing/visual-testing-reporter'),
});
```

> You can specify the `outputFile`

Reviewer: Is there a default? If so, you can mention it. Else, this is not a suggestion ;)

```typescript
export default defineConfig({
  reporter: [
    [require.resolve('@o3r/testing/visual-testing-reporter'), { outputFile: 'visual-testing-report.json' }]
  ]
});
```

## Limitations

> The reporter will contains only the failed evaluated `toHaveScreenshot`.

Reviewer: ```suggestion
The reporter will only contain the screenshots for the failed `toHaveScreenshot` evaluations.
```
